apps/worker/src/app/workflow/usecases/send-message/send-message.usecase.ts

```diff
@@ -97,60 +108,47 @@ export class SendMessage {
         ExecuteOutput<IChimeraChannelResponse> | null
       >({
         ...command,
-        variables: shouldRun.variables,
+        variables: variables,
       });
     }
 
-    if (!command.payload?.$on_boarding_trigger) {
-      const usedFilters = shouldRun?.conditions.reduce(ConditionsFilter.sumFilters, {
-        filters: [],
-        failedFilters: [],
-        passedFilters: [],
-      });
-
-      const digest = command.job.digest;
-      let timedInfo: any = {};
-
-      if (digest && digest.type === DigestTypeEnum.TIMED && digest.timed) {
-        timedInfo = {
-          digestAtTime: digest.timed.atTime,
-          digestWeekDays: digest.timed.weekDays,
-          digestMonthDays: digest.timed.monthDays,
-          digestOrdinal: digest.timed.ordinal,
-          digestOrdinalValue: digest.timed.ordinalValue,
-        };
-      }
+    const { filterResult, channelPreferenceResult } = await this.getStepExecutionHalt(
+      resonateResponse?.options?.skip,
+      command,
+      variables
+    );
 
-      /**
-       * userId is empty string due to mixpanel hot shard events.
-       * This is intentional, so that mixpanel can automatically reshard it.
-       */
-      this.analyticsService.mixpanelTrack('Process Workflow Step - [Triggers]', '', {
-        workflowType: resonateResponse?.outputs ? 'ECHO' : 'REGULAR',
-        _template: command.job._templateId,
-        _organization: command.organizationId,
-        _environment: command.environmentId,
-        _subscriber: command.job?._subscriberId,
-        provider: command.job?.providerId,
-        delay: command.job?.delay,
-        jobType: command.job?.type,
-        digestType: digest?.type,
-        digestEventsCount: digest?.events?.length,
-        digestUnit: digest && 'unit' in digest ? digest.unit : undefined,
-        digestAmount: digest && 'amount' in digest ? digest.amount : undefined,
-        digestBackoff: digest?.type === DigestTypeEnum.BACKOFF || (digest as IDigestRegularMetadata)?.backoff === true,
-        ...timedInfo,
-        filterPassed: shouldRun,
-        preferencesPassed: preferred,
-        ...(usedFilters || {}),
-        source: command.payload.__source || 'api',
-      });
+    if (!command.payload?.$on_boarding_trigger) {
+      this.sendProcessStepEvent(
+        command,
+        resonateResponse?.options?.skip,
+        filterResult,
+        channelPreferenceResult,
+        !!resonateResponse?.outputs
+      );
     }
 
-    if (!shouldRun?.passed || !preferred) {
+    if (!filterResult?.passed || !channelPreferenceResult || resonateResponse?.options?.skip) {
       await this.jobRepository.updateStatus(command.environmentId, command.jobId, JobStatusEnum.CANCELED);
 
-      return;
+      const echoSkip = resonateResponse?.options?.skip;
+      await this.executionLogRoute.execute(
```

Comment:
before we created an execution log only if the filter did not pass, now we will log execution detail on every step halt.